Exceptions.h

```diff
@@ -15,47 +15,47 @@ namespace antlr4 {
     std::string _message;
   public:
     RuntimeException(const std::string &msg = "");
-
+    virtual ~RuntimeException() override;
```

Comment:
I don't understand this part. Why do we need to define default destructor?
Compiler should generate it automatically in every translation unit that includes this file and then linker will merge duplicates.

Reply:
This fixes missing vtable / typeinfo errors. Probably, some macOS-specific behavior of the linker. It also produces a note like this:

NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.